### PR TITLE
Create PawnFi

### PR DESCRIPTION
**Pull request by pawnfidev**
# Grant Application Checklist

- [ ] The [application template](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.

- [ ] A BTC or Ethereum (DAI) address for the payment of the milestones is provided inside the application.  
- [ ] I have read and acknowledged the [Terms and Conditions](https://github.com/w3f/Open-Grants-Program/blob/master/src/T&Cs.md).
- [ ] The software delivered for this grant will be released under an open-source license specified in the application.
- [ ] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants.
- [ ] The initial PR contains only one commit (squash if needed before submitting your PR).
- [ ] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery).

### CODE REVIEW

**Comment by semuelle** [comment]
Thanks for the application, @pawnfidev. We will look into it as soon as possible.

Just curious, your website states PawnFi is live on Ethereum and Binance, but I only see a link to the test app. Is it actually live? On mainnet or a testnet?
Also, I don't see any blockchain or software development experience in your Github or LinkedIn profiles. Could you provide any? Are there any PawnFi parts open sourced yet?

**Comment by pawnfidev** [comment]
> Thanks for the application, @pawnfidev. We will look into it as soon as possible.
> 
> Just curious, your website states PawnFi is live on Ethereum and Binance, but I only see a link to the test app. Is it actually live? On mainnet or a testnet?

Pawnfi Testnet V2 is live on Ethereum Ropsten testnet and Binance Smart Chain testnet, the test link is https://test-app.pawnfi.com. Please log in via the test IDs to verify our demo product. Our next test version will be live in May, and Pawnfi public version is scheduled to be launched after July this year.

> Also, I don't see any blockchain or software development experience in your Github or LinkedIn profiles. Could you provide any? Are there any PawnFi parts open sourced yet?

Please check the profiles of our other core tech-team members as follows:
Dirk Xu
• https://github.com/dirk-pawnfi
• CTO of Pawnfi, former tech leader at Yintech (NASDAQ:YIN)
• Has corporate experience in Fintech softwares development for more than 15 years
• Knowledgeable in distributed service oriented architecture, big data processing and massive service
• Proficient in Solidity & Rust & Golang & C++

Pawnfi has not been open-sourced yet, but if you need we can grant access for you to review it.

Please send an email to dev@pawnfi.com via your email of github account. And we will send the test ID and grant access of the private repo to you.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sorry for the late response. Could you integrate this information into the application? You can give me (https://github.com/Noc2) access to your repositories. I also just sent you an email regarding the test id. Regarding Non-Standard Assets, to which definition are you referring here, because ERC20, ERC721 and ERC1155 are standards? What exactly are the functions of “crowd lending” and the marketplace? Are you aware of acala (https://github.com/AcalaNetwork/Acala/)? How is your solution different from them? And are you aware of the [asset module](https://github.com/paritytech/substrate/tree/master/frame/assets)?

**Comment by pawnfidev** [comment]
Hi David,

> Sorry for the late response. Could you integrate this information into the application? 
- Done

> You can give me (https://github.com/Noc2) access to your repositories. 
- Done

> I also just sent you an email regarding the test id. 
- Replied with the test id
 
> Regarding Non-Standard Assets, to which definition are you referring here, because ERC20, ERC721 and ERC1155 are standards? 

While standard for ERC20, ERC721 and ERC1155 are actually the rules/ identifiers that how these tokens are created, the standard we are talking about is actually homogeneity. That is, all kinds of customized products or tokens are considered as Non-Standard Assets. No matter it is NFTs, LP certificates, or even derivatives, insurance policy, while the nature of assets are not the same among different holders, those assets are not “Standard” at all and PawnFi sees the pain point of assets holders: lack of price fetching mechanism and low utility.

In sum, no matter it is ERC20, ERC721 or ERC1155, or even the derivatives that built on top of it, PawnFi is acting as a solution to unleash its value, improve its turnover rate, and boost circulation as a whole.

> What exactly are the functions of “crowd lending” and the marketplace? 

In our crowd lending module, we developed two models: FCFS and Subscription. The aim of crowd lending is for NSAs which few reference prices on the market. Such NSA normally does not have high turnover rate in the market. By using crowd lending, Pawnfi can release the liquidity and improve visitability. For example, for the items in certain games, which never be collateralized or traded before, crowd lending can help owners to achieve the target fund raising outside of the gaming world. The interested participants can decide how much do they want to fund in this items and gain interest for returns. In short, crowd lending opens the border of collateralized loan and build a linkage between DeFi and NFT players.

To us, the marketplace is applied in many scenarios. From time to time, users may want to keep the ownership while lease out their usability to maximize the utility of their NSA. For example, instead of let LP tokens sit the wallet, owners can lease out their LP tokens and earn interests while lessee can participate certain LP events. Or users can stake their NSA in exchange for collateral loan that they can apply to their short term funding needs, while keep the ownership of NSA without worrying about ownership.

> Are you aware of acala (https://github.com/AcalaNetwork/Acala/)? How is your solution different from them? 

We heard about it but not very much. We focus on NSA which is the biggest difference.

> And are you aware of the [asset module](https://github.com/paritytech/substrate/tree/master/frame/assets)?

Yes, but in Pawnfi Chain, we are working on the lending/leasing/selling for different assets like a pawnshop, not only just support asset types.

Hope our description is clear for you, if you have any question on PawnFi, feel free let us know.

Thanks

**Comment by Noc2** [comment]
Thanks for the additional information. I just looked at the repository you shared with me, but it actually doesn’t contain any of the mechanisms you plan to implement as part of the grant as far as I can see it. At least I didn’t find anything related to lending, leasing or marketplace in the repo. It mostly seems to be based on some old sushi swap/Chef Nomi code. Could you integrate more information about the functions of lending, leasing or marketplace into the application itself? Did you already previously work on any rust/substrate based projects? 
The UI, which you shared definitely looks nice. Did you develop it yourself and would it be also part of this grant?

**Comment by semuelle** [comment]
> Hi, David, we just updated a README file in the repo, and you can find the code structure there. We think it will help you know better about the mechanisms.

Which repo is this? Your Github org shows no public repos.

> It's fully developed by ourselves, and it's part of the grant as well. But it will not be open sourced in the future. 

I don't see a UI in your deliveries, but any work that is funded by the Web3 Foundation Grants Program must be open-sourced.

**Comment by pawnfidev** [comment]
> > Hi, David, we just updated a README file in the repo, and you can find the code structure there. We think it will help you know better about the mechanisms.
> 
> Which repo is this? Your Github org shows no public repos.
> 
> > It's fully developed by ourselves, and it's part of the grant as well. But it will not be open sourced in the future.
> 
> I don't see a UI in your deliveries, but any work that is funded by the Web3 Foundation Grants Program must be open-sourced.

Hi, team
About the github repo and test product, we have shared with @Noc2 
The code will be open sourced after the product go live

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sorry for the very late reply and thanks for the additional information. It’s now a little bit clearer to me, but it’s still hard to follow and understand the benefits and details of your protocol based on your application. Especially for the rest of the committee, since they don’t have access to your github repository. For example what exactly will you implement, with each module? I personally also think that 27k for the development of the substrate modules with so little details and no visible prior experience with substrate is difficult to justify.  I personally would be mostly interested in the UI at this stage, but this isn’t part of the deliveries at this stage and it seems you don’t want to open source it. Nevertheless I will already share the application with the rest of the committee.

**Comment by semuelle** [comment]
@pawnfidev, could you provide a list of functions you are planning to implement in Deliverables 1-4 and a description of what exactly you want to do for D5?

**Comment by pawnfidev** [comment]
> Sorry for the very late reply and thanks for the additional information. It’s now a little bit clearer to me, but it’s still hard to follow and understand the benefits and details of your protocol based on your application. Especially for the rest of the committee, since they don’t have access to your github repository. For example what exactly will you implement, with each module? I personally also think that 27k for the development of the substrate modules with so little details and no visible prior experience with substrate is difficult to justify. I personally would be mostly interested in the UI at this stage, but this isn’t part of the deliveries at this stage and it seems you don’t want to open source it. Nevertheless I will already share the application with the rest of the committee.

Hi @Noc2 @semuelle  Thanks for your suggestions. Our team discussed it carefully and re-evaluated the workload. We updated a version of the deliveries and add more details of features.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. I have a few additional comments/questions:

- Could you explain the following: “Pawnfi Chain will differentiate each identity within the encrypted network in order to improve transaction throughput and speed, and reduce transaction fees”? 
- Could you explain why you want to use substrate modules? Maybe an ink! Smart contract would make more sense, especially since it seems you're deploying the same solution on different networks and you won’t optimize it.  
- I still think that the application is rather expensive, also given that you’re deploying it on multiple networks. 
- Could you compare your solution to Aave? Or are there any other similar solutions on ethereum? 
- Are you somehow related to following team: https://pawn.fi/

**Comment by pawnfidev** [comment]
> Thanks for the update. I have a few additional comments/questions:
> 
>     * Could you explain the following: “Pawnfi Chain will differentiate each identity within the encrypted network in order to improve transaction throughput and speed, and reduce transaction fees”?
The purpose of Pawnfi chain is to achieve functions such as leasing, lending and borrowing, appraisement, repayment and etc of Nonstandard asset on chain. Compared to smart contract, deployed the features on chain can have improved efficiency and lower gas fee to support our reliance on crowd intelligence.

>     * Could you explain why you want to use substrate modules? Maybe an ink! Smart contract would make more sense, especially since it seems you're deploying the same solution on different networks and you won’t optimize it.
All the available functions of Pawnfi are developed and tested rigorously based on EVM. For deployment purpose and long-term planning, we hope the functions we mentioned above (such as leasing, lending and borrowing, appraisement, repayment and etc) can be deployed on substrate modules and applied on other future projects as well. While using digital asset as collateral is a natural step for completed ecosystem, the readiness of PawnFi will shorten the development period and increasing the efficiency among other projects.

 
>     * I still think that the application is rather expensive, also given that you’re deploying it on multiple networks.
So far, our progress is steady and as expected. Pawnfi has already trusted by some investors and we have received their capitals for us to develop on other network. We hope we can get web3 support for us to expediate the develop based on Substrate. 

>     * Could you compare your solution to Aave? Or are there any other similar solutions on ethereum?

First of all, while Aave, Compound and MakerDAO are focusing only on Collateral Loan for main stream coin, PawnFi’s vision is to unleash the liquidity for all the non-Standard asset like NFTs, Financial Derivatives or even LP certificate. By introducing different kinds of lending module such as Pool lending, Crowd Lending and Appraisement, we are able to target asset with different turnover rates. That is, Pawnfi not only can handle all types of mainstream assets lending that Aave/Compound can do in their pool lending, but also be equipped with the Non-standard-assets lending at Aave/Compound cannot realize in their current product form.
 
Secondly, many of the parameters in Pawnfi can be customized by the borrower. In addition to the flexibility of collateral assets, the borrowing period, interest rate and even repayment method can be freely selected or combined by the borrower. As we believe that the more flexibility we provide to the lending agreement, the more liquidly we can create for all types of assets.
 
Thirdly, not only for lending, one of the core features in PawnFi is Leasing. The current pain point for many market participants is that, how to create short term liquidity and increase asset turnover rate. For existing players, selling asset to gain liquidity means giving up potential gains; for new joiners, high entry cost due to soaring NFT prices is forming an invisible barrier. In this case, Pawn shop that equips with loan and leasing market is the best solution. While our inhouse module enables all kinds of assets with different turnover rate to be placed for collateral loan to create liquidity need from anyone in the market, leasing market not only serves as a Stepping stone for new players to get started w/o high entry cost, but increases turnover rate of Non-standard assets among the whole market. In sum, shot term liquidity generation and stable return will no longer be exclusive but can happen in parallel.
 
Last but not least, unlike other platforms that is limited to either Defi or NFT players, Pawnfi enables the broader asset types and is capable of grabbing the market shares.
Combined our different lending module like crowd lending/ pool lending and appraisement with leasing market, Pawnfi is becoming the fair price source for NSAs with low turn over rate. In long term, the cobination of Lending, Leasing and even our resale market in whole PawnFi ecosystem will attract core players not only asset holders, but also fund holders from Defi, who are rich in resources and capable of pricing asset to participate in liquidity generation and asset transfer.
 
 
Other similar solutions on Ethereum are like NFTX and NFTfi. However, PawnFi view differently on how the liquidity should be generated.
 
First of all, unlike other NFT launching and trading platform that is limited to NFT players, for Assets which is not suitable for high frequency trading, we believe Lending and leasing is the best way to fetch liquidity and bring in Audience from the whole blockchain.
 Secondly, Non-Standard asset lending should not be P to P but should be P to crowd. As many of NSAs like NFTs are not common or with low turnover rate in the market, it is hard and inefficient to find a willing lender to match a random/ specific loan contract. For this reason, while Pool lending is too risky to define those lending parameters, Crowd Lending should be the best solution to such case. Via collective intelligence and willingness to lend, we can make the NSA collateral loan more efficient and more referable to forward lending practice in the market.
 
Thirdly, NFT should not always be divided into pieces. Unlike stocks and mainstream coins that its value can be easily identified, the intrinsic value of NFTs or NSAs often comes from its use case and meaning as a whole. Other than that, many of the NFT or NSA holders do not want to sell it but generate short term liquidity and asset use case because selling it means giving up long term capital gain. For this reason, a market combined with Lending and leasing will most likely meet their need and broaden their use case, while at the same time encourage them to hold their asset.

>     * Are you somehow related to following team: https://pawn.fi/
 No we are not related at all, just happen to have the similar naming.

**Comment by Noc2** [comment]
Thanks for the reply and detailed explanation. I will share your application with the rest of the committee. But  I’m still not sure how “identity” is related to “improve transaction throughput and speed, and reduce transaction fees”. I understand that there are advantages of using a pallet compared to smart contracts, but that doesn’t improve for example the “speed” of transactions. Also your explanation regarding smart contracts vs module, sounds like it would be better for you to actually rely on the existing solidity implementations and for example to use the evm module (see for example https://github.com/PureStake/moonbeam).

**Comment by pawnfidev** [comment]
> Thanks for the reply and detailed explanation. I will share your application with the rest of the committee. But I’m still not sure how “identity” is related to “improve transaction throughput and speed, and reduce transaction fees”. I understand that there are advantages of using a pallet compared to smart contracts, but that doesn’t improve for example the “speed” of transactions. Also your explanation regarding smart contracts vs module, sounds like it would be better for you to actually rely on the existing solidity implementations and for example to use the evm module (see for example https://github.com/PureStake/moonbeam).

Sorry for the confusion regarding to “identity” and “improve transaction throughput and speed, and reduce transaction fees”. Please discard these and allow me to rephrase our future plan here. 
The purpose of Pawnfi chain is to achieve functions such as leasing, lending/borrowing, appraisement, repayment and etc. of Non-standard assets on chain. Comparing to smart contract, deploying these functions on Pawnfi chain can help us improve efficiency, lower gas fee and ultimately greatly benefit from crowd intelligence.

Another thing is, we have been working with Moonbeam for a while and our EVM version has already been rigorously running on Moonbeam Testnet. There is actually no conflict between EVM and substrate module.

**Comment by mmagician** [comment]
At the current state of the proposal I will not be accepting this application, for the following reasons:
- There is no demonstrable coding experience. The github profile you shared is practically empty and according to @Noc2 the private repo doesn't contain related code
- The price is too high for the proposed scope. Some of the NSAs you mention don't yet exist in our ecosystem (such as LP certificates), so I'm not sure how you would cater for these in your design right now.
- To the best of my knowledge, your definition of p2p vs crowd lending is mixed up and you're hence contradicting yourself here:
> As many of NSAs like NFTs are not common or with low turnover rate in the market, it is hard and inefficient to find a willing lender to match a random/ specific loan contract. For this reason, while Pool lending is too risky to define those lending parameters, Crowd Lending should be the best solution to such case.

Actually pool lending is much better suited for environments with low liquidity, because you don't need to find an explicit buyer-seller pair for each trade.

**Comment by mmagician** [comment]
As this application has been open for almost 3 months and hasn't received any approvals, I am closing it for now. Feel free to apply for another project in the future and thank you for your interest in the Web3 Grants Program.

**Comment by mmagician** [comment]
As this application has been open for almost 3 months and hasn't received any approvals, I am closing it for now. Feel free to apply for another project in the future and thank you for your interest in the Web3 Grants Program.
